gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc

```diff
@@ -54,9 +86,26 @@ int main(int argc, char* argv[]) {
     return 1;
   }
 
+  const std::string json_output_dir = absl::GetFlag(FLAGS_json_output_dir);
   const std::string output_dir = absl::GetFlag(FLAGS_output_dir);
-  if (output_dir.empty()) {
-    std::cerr << "Error: --output_dir is required\n";
+  const std::vector<std::string> output_types =
+      absl::GetFlag(FLAGS_output_types);
+  bool json_output_enabled = false;
+  bool asm_output_enabled = false;
+  for (const std::string& output_type : output_types) {
+    if (output_type == "json")
+      json_output_enabled = true;
+    else if (output_type == "asm")
+      asm_output_enabled = true;
+  }
+
+  if (json_output_enabled && json_output_dir.empty()) {
+    std::cerr << "Error: --json_output_dir is required when the json output "
+                 "type is requested\n";
+    return 1;
+  } else if (asm_output_enabled && output_dir.empty()) {
```

Comment:
ok, so you're only supporting one output. so might as well type the flag as std::string (or an enum and self-validate)

Reply:
This was meant to be another `if` statement. Sorry for the confusion.

I've switched to just looking at the output directories and have removed this code.